### PR TITLE
Include attr/event callbacks when resubscribing a paired node.

### LINKED ISSUE #1121: Paired Node looses attribute callbacks on re-subscribing
Hi, I noticed that after a few days of running a modified version of the shell app some nodes would send attribute changed messages, but my callbacks were not being called. I tracked this down to this line:

https://github.com/project-chip/matter.js/blob/81c0c32f0c2abf360aceb980189bd7387ef2dfc2/packages/matter.js/src/device/PairedNode.ts#L375-L378

which is subscribing again when it detects a missed subscription event, but is not passing in the callbacks originally setup with the paired node.  

I have a PR that fixes this for me and maybe better describes the issue, but feel free to disregard/close it if there is a better solution ;-)

### PR DESCRIPTION
**Pull request by digitaldan**
Fixes #1121

### CODE REVIEW

File: packages/matter.js/src/device/PairedNode.ts [hunk 1]
```diff
@@ -374,7 +374,12 @@ export class PairedNode {
             updateTimeoutHandler: async () => {
                 logger.info(`Node ${this.nodeId}: Subscription update not received ...`);
                 try {
-                    await this.subscribeAllAttributesAndEvents();
+                    const { attributeChangedCallback, eventTriggeredCallback } = this.options;
+                    await this.subscribeAllAttributesAndEvents({
+                        ignoreInitialTriggers: true,
+                        attributeChangedCallback: data => attributeChangedCallback?.(this.nodeId, data),
```
Comment (Apollon77) [inline packages/matter.js/src/device/PairedNode.ts:380]: Why you change here and next line the callback method? In fact what's missing is thatched original options are also set when resubscribing. So passing the callbacks from options just like in options of the subscribeAllAttributesAndEvents method should be sufficient or?

File: packages/matter.js/src/device/PairedNode.ts [hunk 2]
```diff
@@ -374,7 +374,12 @@ export class PairedNode {
             updateTimeoutHandler: async () => {
                 logger.info(`Node ${this.nodeId}: Subscription update not received ...`);
                 try {
-                    await this.subscribeAllAttributesAndEvents();
+                    const { attributeChangedCallback, eventTriggeredCallback } = this.options;
+                    await this.subscribeAllAttributesAndEvents({
+                        ignoreInitialTriggers: true,
```
Comment (Apollon77) [inline packages/matter.js/src/device/PairedNode.ts:379]: That's a good question depending on how long it stayed disconnected we could miss data. So better miss them all as change callbacks or get change-calls that are for the same value. Maybe second is better or? Better more then missing? Then this should be false.

File: packages/matter.js/src/device/PairedNode.ts [hunk 3]
```diff
@@ -374,7 +374,12 @@ export class PairedNode {
             updateTimeoutHandler: async () => {
                 logger.info(`Node ${this.nodeId}: Subscription update not received ...`);
                 try {
-                    await this.subscribeAllAttributesAndEvents();
+                    const { attributeChangedCallback, eventTriggeredCallback } = this.options;
```
Comment (Apollon77) [inline packages/matter.js/src/device/PairedNode.ts:377]: This.options or better using what comes in in this method as params?

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Hey, I commented ... I am just on mobile currently so I might not have the full picture of the file in my head right now ... but I hope my comments make sense.

File: packages/matter.js/src/device/PairedNode.ts (hunk 3 above)
Comment (digitaldan) [reply packages/matter.js/src/device/PairedNode.ts:377]: yeah, i totally spaced this was actually calling itself and already had the options  passed in.

File: packages/matter.js/src/device/PairedNode.ts (hunk 1 above)
Comment (digitaldan) [reply packages/matter.js/src/device/PairedNode.ts]: Yeah, my oversight.

**Review by Apollon77 — APPROVED** [review]

**Comment by Apollon77** [comment]
Thank you. Fix will be in next nightly tomorrow.

**Pull request merged**